src/ecore.js

```diff
@@ -861,6 +868,9 @@ EClass.values = {
 
         return _.isArray(subTypes) ? subTypes : [];
     },
+    hasSubTypes: function () {
```

Comment:
One of the questions we wanted to ask is would there be a simpler way to detect this instead of getting the eAllSubTypes length multiple times.  We were thinking that it might be possible to detect potential subTypes on loading of the ecore file such that the subsequent calls would be much more efficient.  Just a question...  That is why we added this derived attribute over here although we do agree that it does not match the underlying EMF implementation...